src/ofxstatement/parser.py

```diff
@@ -53,7 +53,7 @@ def split_records(self) -> Iterable[LT]:  # pragma: no cover
         """Return iterable object consisting of a line per transaction"""
         raise NotImplementedError
 
-    def parse_record(self, line: LT) -> Optional[StatementLine]:  # pragma: no cover
+    def parse_record(self, line: LT) -> Optional[StatementLine|InvestStatementLine]:  # pragma: no cover
```

Comment:
Hm, I somehow missed the moment when this `InvestStatementLine` was introduced (found it now in #129), but I'm not a big fan of these ambiguous return type.

What typing check errors are you trying to address with this?

Reply:
`error: Return type "InvestStatementLine | None" of "parse_record" incompatible with return type "StatementLine | None" in supertype "StatementParser"  [override]`

I'm building a plugin to parse a JSON file filled with investment information, and mypy is throwing an override error due to the fact that my declaration says I'm returning an InvestStatementLine.

I neglected to notice that the `|` syntax was introduced in 3.10, and I think I used it incorrectly anyways. I can fix the PR, but do you want to approach this a different way? I can always have it ignore that specific type check in my parser code, but I thought it was better to address